From 90cece5afe8062ef944deffd9a06c4d37443b234 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Wed, 2 Oct 2019 16:06:47 +0200
Subject: [PATCH] Fixed issue #1709 bad type break bigendian build

---
 php_xdebug.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/php_xdebug.h b/php_xdebug.h
index 7d5bfab53..0002f2ea3 100644
--- a/php_xdebug.h
+++ b/php_xdebug.h
@@ -253,7 +253,7 @@ ZEND_BEGIN_MODULE_GLOBALS(xdebug)
 	zend_bool     remote_connect_back;   /* connect back to the HTTP requestor */
 	char         *remote_log;       /* Filename to log protocol communication to */
 	FILE         *remote_log_file;  /* File handler for protocol log */
-	int           remote_log_level; /* Log level XDEBUG_LOG_{ERR,WARN,INFO,DEBUG} */
+	zend_long     remote_log_level; /* Log level XDEBUG_LOG_{ERR,WARN,INFO,DEBUG} */
 	zend_long     remote_cookie_expire_time; /* Expire time for the remote-session cookie */
 	char         *remote_addr_header; /* User configured header to check for forwarded IP address */
 	zend_long     remote_connect_timeout; /* Timeout in MS for remote connections */
